experiments/ClimaEarth/setup_run.jl

```diff
@@ -97,16 +97,6 @@ function CoupledSimulation(config_file = joinpath(pkgdir(ClimaCoupler), "config/
 end
 
 function CoupledSimulation(config_dict::AbstractDict)
-    # Make a copy so that we don't modify the original input
-    config_dict = copy(config_dict)
```

Comment:
We need this, otherwise we cannot run the same simulation twice in the same session (as we do in the restarts).

Reply:
I think I had problems with `parse_component_dts!(config_dict)`, because it changes the dict

Reply:
I think that @Sbozzolo is saying that, although it may work, it's not easy to reason about because `Dict`s are passed by reference, so removing the `copy` makes this a mutating method. If we did remove the `copy`, then we should change `CoupledSimulation` to `CoupledSimulation!`. I'm not exactly sure how this method mutates the dict, but we would want `CoupledSimulation!` to be idempotent for it to be used correctly in that way.

Reply:
Keeping the `copy` makes it simpler in this way.

Reply:
In this PR I changed the `CoupledSimulation` function so it doesn't modify `config_dict`. Now, the config dict is created in `get_coupler_config_dict`, and is not modified outside of that function. So as long as we use the config dict output from `get_coupler_config_dict`, our runs will be reproducible.

Previously, the restart test was manually parsing the config dict, duplicating what's done in `get_coupler_config_dict`. Now it just calls that function directly, so on subsequent runs we can call `get_coupler_config_dict(config_file)` and reproduce the config dict that was used